src/main/java/app/controllers/ProfessorController.java

```diff
@@ -74,21 +87,77 @@ public ModelAndView saveProject(@AuthenticationPrincipal UserDetails currentUser
+    /**
+     * Navigate to the Add Student Page
+     * @param model Model
+     * @param projectID Project ID to link Student
+     * @return Redirect to the "new-student" page
+     */
+    @RequestMapping("/new-student/{projectID}")
+    public ModelAndView addStudentById(Model model, @PathVariable("projectID") Long projectID)
     {
+        model.addAttribute("students", studentRepository.findAll());
+        model.addAttribute("projectID", projectID);
         return new ModelAndView("new-student","command", new Student());
     }
 
 
-    // Not currently working. Need to change the confirmation workflow
+    /**
+     * Save the Student to the Project
+     * @param currentUser Logged in user
+     * @param id Student's ID
+     * @param projectID Project's ID
+     * @return Redirect to the "facultyMenu" page
+     */
     @RequestMapping(value = "/save-student", method = RequestMethod.POST)
-    public ModelAndView addStudent(@AuthenticationPrincipal UserDetails currentUser, @RequestParam("id") Long id)
+    public ModelAndView addStudent(@AuthenticationPrincipal UserDetails currentUser,
+                                   @RequestParam("id") Long id,
+                                   @RequestParam("projectID") Long projectID)
     {
         AuthenticatedUser authenticatedUser = authenticatedUserService.findByUsername(currentUser.getUsername());
         Professor professor = authenticatedUser.getProfessor();
         student = studentRepository.findOne(id);
+        project = projectRepository.findOne(projectID);
+
+        project.addStudent(student);
+        professorRepository.save(professor);
+
+        return new ModelAndView("redirect:/facultyMenu");
+    }
+
+    /**
+     * Navigate to the Add Second Reader page
+     * @param model Model
+     * @param projectID Project ID to link Second Reader
+     * @return Redirect to "add-reader" page
+     */
+    @RequestMapping("/add-reader/{projectID}")
+    public ModelAndView addReaderById(Model model, @PathVariable("projectID") Long projectID)
+    {
+        model.addAttribute("readers", professorRepository.findAll());
+        model.addAttribute("projectID", projectID);
+        return new ModelAndView("add-reader","command", new Professor());
+    }
+
+    /**
+     * Save Second Reader to the Project
+     * @param currentUser Logged in user
+     * @param id Second Reader's ID
+     * @param projectID Project's ID
+     * @return Redirect to the "facultyMenu" page
+     */
+    @RequestMapping(value = "/save-reader", method = RequestMethod.POST)
+    public ModelAndView addReader(@AuthenticationPrincipal UserDetails currentUser,
+                                  @RequestParam("id") Long id,
+                                  @RequestParam("projectID") Long projectID)
+    {
+        AuthenticatedUser authenticatedUser = authenticatedUserService.findByUsername(currentUser.getUsername());
+        Professor professor = authenticatedUser.getProfessor();
```

Comment:
Add a redirect to the facultyMenu if the project isn't in the list of projects for the projessor

Reply:
Ditto!